engine/rpcserver_test.go

```diff
@@ -266,11 +265,7 @@ func RPCTestSetup(t *testing.T) *Engine {
 	if err != nil {
 		t.Fatal(err)
 	}
-	tempDir, err := ioutil.TempDir("", "")
-	if err != nil {
-		t.Fatal(err)
-	}
-	dbm.dbConn.DataPath = tempDir
+	dbm.dbConn.DataPath = t.TempDir()
```

Comment:
Unfortunately, I'm encountering issues with this:
engine package:
```
 testing.go:1090: TempDir RemoveAll cleanup: remove C:\Users\gloriousCode\AppData\Local\Temp\TestCheckConnection2059912713\001\test.db: The process cannot access the file because it is being used by another process.
```
or in the config package:
```
--- FAIL: TestEncryptTwiceReusesSaltButNewCipher (2.48s)
    testing.go:1090: TempDir RemoveAll cleanup: remove C:\Users\gloriousCode\AppData\Local\Temp\TestEncryptTwiceReusesSaltButNewCipher1474172188\001\455337499.pw: The process cannot access the file because it is being used by another process.
```

Or for AppVeyor, one of our CI's: https://ci.appveyor.com/project/thrasher-/gocryptotrader-vxsws/builds/43348807#L7709

I was initially receiving it for the test `TestRPCServer_GetTicker_LastUpdatedNanos`, which I updated to no longer use `RPCTestSetup`, but then as above, the test `TestCheckConnection` encountered an issue

While I love that a new function has been created for this in Golang, there is some complexity in `engine` package with go routines and such inside tests which may mean we may not be able to use this. I'll need to do further investigation

Reply:
I see that you are using a Windows machine. Windows does not permit removing an open file, so if we didn't close an `os.File` properly, then `t.TempDir()` will not be able to perform cleanup
```go
func TestFoo(t *testing.T) {
	tmpDir := t.TempDir()

	file, err := os.CreateTemp(tmpDir, "dumpwaltest*.txt")
	assert.NoError(t, err)

	// On Windows, the test will fail if `file` is not closed because
	// t.TempDir() will not be able to cleanup the temporary directory.
	//
	// Unix permits removing an open file, Windows does not.
	t.Cleanup(func() {
			file.Close()
	})
}
```

I don't have a Windows machine myself, so I'm going to use the GitHub-hosted Windows runner on my fork to run the tests. I'll push new commits when it's done